### PR TITLE
119 events

### LINKED ISSUE #119: [FEATURE] - Events
**Issue Description**
Suggested destinations are also depending on events that might occur on/near airports, such as aeronautical events, meetings, fly ins, ...

Therefore we need to add a new Model/Controller/view called Events.

**table structure**
- airport_id
- type
- start_date
- end_date
- image_link
- url

Events should be visible by signed in users.
Events should be created/updated only by users having the role `admin`.

Users can access on events (READ) on index and show.

**TODO LIST**

- [x] Create Model Event
- [x] Create Controller EventsController
- [x] Create the create view for admins
- [x] Create the update/delete view for admins
- [x] Create access to events searcher in the nav bar under `Search > Events`
- [x] Adapt the `Destination.rb/get_top_destinations` in order to promote airports that offer an event during the correct time frame (of course if airport in flyzone)
- [x] Create events seeds for development
- [x] Write tests

**linked issues**
Issue #120 should be closed before!

**Open questions**
- [ ] Should users be able to comment on events? I think yes...

### PR DESCRIPTION
**Pull request by christinasugiono**
closes #119 

Events
image

Note: Edit button only visible to `admin`
image

New and Edit Form (Delete action is inside edit form)
image

### CODE REVIEW

File: app/views/events/_events.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<%# locals: (events:) %>
```
Comment (christinasugiono) [inline app/views/events/_events.html.erb:1]: this line is about strict locals, which is better and safer when using partials

https://masilotti.com/safer-rails-partials-with-strict-locals/

File: app/views/events/_events.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      <% if event.image_link.present? %>
+        <%= image_tag event.image_link, alt: event.title, class: "card-image" %>
+      <% else %>
+        <%= image_tag "https://sman1ngantang.sch.id/wp-content/plugins/elementor/assets/images/placeholder.png", alt: event.title, class: "card-image" %>
+      <% end %>
+      <div class="card-trip-infos">
+        <h2><%= event.title %></h2>
+        <p class="mt-5"><%= event.start_date.strftime("%a, %e %b") %><%= "- #{event.end_date.strftime("%a, %e %b")}" unless event.one_day? %></p>
+        <div class="d-flex align-items-center mt-10 justify-content-between">
+          <p class="mt-0">📍 <%= "#{event.airport.name} (#{event.airport.icao})" %></p>
```
Comment (alexstan67) [inline app/views/events/_events.html.erb:22]: Just a last update. Something you cannot know. Not all airports have a "real" ICAO code. For now, we just display the real ones.
So we display the parenthesis and content **only** if:
`event.airport.icao.match?(/^(L|E)...$/)`

**Review by alexstan67 — CHANGES_REQUESTED** [review]

**Comment by alexstan67** [comment]
Hi @christinasugiono, just a minor change (display only real ICAO codes), result looks amazing! Once you fix it, if you have the rights, you can merge to master!

**Comment by christinasugiono** [comment]
image

Last push 💪

**Review by alexstan67 — APPROVED** [review]

**Comment by alexstan67** [comment]
To continue with good practices, who has to merge this branch on `master`, me or you?

**Pull request merged**